### ISSUE TITLE
Face::set_variation change in behaviour

### ISSUE THREAD

**Issue by alexheretic** [RazrFalcon/ttf-parser#134]
I noticed `Face::set_variation` now returns `Some(())` even when the face does not have the axis since v0.20.0.

If this is intended the docs should be amended.

> Returns `None` when face is not variable or doesn't have such axis.

## Reproduce
```rust
// [dependencies]
// ttf-parser19 = { version = "0.19", package = "ttf-parser" }
// ttf-parser = { version = "0.20" }

const VAR_FONT: &[u8] = include_bytes!("../Cantarell-VF.otf");

fn main() {
    let mut face = ttf_parser19::Face::parse(VAR_FONT, 0).unwrap();
    dbg!(face.set_variation(ttf_parser19::Tag::from_bytes(b"wght"), 600.0)); // Some(())
    dbg!(face.set_variation(ttf_parser19::Tag::from_bytes(b"foob"), 200.0)); // None
    dbg!(face.set_variation(ttf_parser19::Tag::from_bytes(b"barr"), 200.0)); // None

    let mut face = ttf_parser::Face::parse(VAR_FONT, 0).unwrap();
    dbg!(face.set_variation(ttf_parser::Tag::from_bytes(b"wght"), 600.0)); // Some(())
    dbg!(face.set_variation(ttf_parser::Tag::from_bytes(b"foob"), 200.0)); // Some(())
    dbg!(face.set_variation(ttf_parser::Tag::from_bytes(b"barr"), 200.0)); // Some(())
}
```

Using [Cantarell-VF.otf](https://github.com/alexheretic/owned-ttf-parser/blob/master/fonts/Cantarell-VF.otf).

**Comment by RazrFalcon** [comment]
Yes, my bad. It should not return anything now. You have to use `face.variation_axes()` to check for available tags beforehand.

This mimics the harfbuzz API: https://github.com/harfbuzz/harfbuzz/blob/b0150ffd05523c7cef9b3c91a8d6050177fceda3/src/hb-font.cc#L2658-L2676

**Comment by Goldziher** [comment]
Fixed in `b58c472` (#207) — `Face::set_variation` now returns `None` when the axis tag is not present on the face.

Worth being explicit that this went the opposite way from the direction Yevhenii leaned in this thread. You offered both fixes ("if this is intended the docs should be amended"), and the reply here favoured amending the docs to keep parity with harfbuzz, where `hb_font_set_variation` returns `void`.

We chose the code fix instead. The rustdoc had promised `None` for an unknown axis for five releases, and callers reasonably relied on the documented contract; harfbuzz parity is weaker justification when the function already returns `Option<()>` rather than `()`. It is a behavioural change with no compile error, so it is called out explicitly under `### Changed` in the 0.26.0 notes rather than filed as a bugfix.

Thanks for the clear report, and sorry it took this long to resolve.
